### ISSUE TITLE
Duplicate validation error on nested property

### ISSUE THREAD

**Issue by freezy** [Automattic/mongoose#4681]
Upgrading to latest mongoose broke my tests. I nailed it down to the [4.6.2 update](https://github.com/Automattic/mongoose/compare/4.6.1...4.6.2) (however the bug is still present in 2.6.6). Basically I'm getting a validation error I shouldn't:

### model.js
```javascript
const mongoose = require('mongoose');
const NestedSchema = new mongoose.Schema({ 
    nestedName: { type: String, required: 'Name is required.' },
    created_at: { type: Date, required: true }
});
const RootSchema = new mongoose.Schema({
    rootName:  String,
    nested: { type: [ NestedSchema ] }
});
mongoose.model('Root', RootSchema);
```

### reproduce.js
```javascript
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost/issue-4681');
const model = require('./model.js');
const Root = mongoose.model('Root');
const root = new Root({ rootName: 'root', nested: [ { } ] });
root.save().then(result => {
    console.log('All okay (dont care)!');
}, err => {
    if (err.errors) {
        for (let path in err.errors) {
            console.log('ERR [%s]: %s', path, err.errors[path]);
        }
    }
});
```

- 4.6.1 outputs: 
   ```
   ERR [nested.0.created_at]: Path `created_at` is required.
   ERR [nested.0.nestedName]: Name is required.
   ```
- 4.6.2 outputs:
   ```
   ERR [nested.0.created_at]: Path `created_at` is required.
   ERR [nestedName]: Name is required.
   ERR [nested.0.nestedName]: Name is required.
   ```

Obviously the second line shouldn't be displayed. Note that **this only happens together with `created_at`**. Removing `created_at` from `NestedSchema` results in only one validation error.

Any idea why this happens?

**Comment by vkarpov15** [comment]
Good catch, fixed in master and fix will be released with 4.6.7

**Comment by freezy** [comment]
Sweet, thanks!
